### ISSUE TITLE
No available api to make talkback or voiceover to spellout characters

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#79318]
b/182520487
b/182336500
b/182338851

## Steps to Reproduce

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    const title = 'code sample';
    return MaterialApp(
      title: title,
      home: Scaffold(
        body: Home(),
      ),
    );
  }
}

class Home extends StatelessWidget {
  const Home({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Text('IwantThisSpellOut'),
          Text('IwantThisSpellOut'),
        ],
      ),
    );
  }
}
```

There is no API in flutter to make the talkback or voiceover to announce character by character if the text receives focus.

Design doc: [flutter.dev/go/a11y-text-attributes](flutter.dev/go/a11y-text-attributes)

**Comment by chunhtai** [comment]
In android, you can set the ttspan with type Verbatim to make talkback spell out the word

example code
```java
SpannableString spannableString = new SpannableString(content);
        TtsSpan ttsSpan = new TtsSpan.Builder(TtsSpan.TYPE_VERBATIM).build();
        spannableString.setSpan(ttsSpan, 0, java.lang.Math.min(2, content.length()), 0 /* no flag */);
        result.setContentDescription(spannableString);
```

**Comment by chunhtai** [comment]
In iOS

example code
```objc
- (NSAttributedString*)accessibilityAttributedLabel {
  NSMutableAttributedString* attributedString = [[NSMutableAttributedString alloc] initWithString:[self accessibilityLabel]];
  if (@available(iOS 13.0, *)) {
    [attributedString setAttributes:@{UIAccessibilitySpeechAttributeSpellOut: @(YES)} range:NSMakeRange(0, attributedString.length)];
  }
  return attributedString;
}
```

**Comment by chinmaygarde** [comment]
@jonahwilliams mentions that a workaround for this in the application could be to provide an alternate semantics label that provides a phonetic string. You could do this by inserting spaces between the characters of the sentence. But @chunhtai seems to be on a correct fix for this.

**Comment by chunhtai** [comment]
@chinmaygarde There are some issues with that approach. First android seems to missed some character when they are separated by the space, see b/182438775. There is also some issue when applying this to iOS, it causes text editing to act weirdly, which is the root cause of https://github.com/flutter/flutter/issues/67754.

On the other hand, this API can be easily extended to support https://github.com/flutter/flutter/issues/53972

**Comment by chinmaygarde** [comment]
👍🏽  Sounds good.

**Comment by chunhtai** [comment]
which internal bug are you referring to? I almost done with the other thing i am working on, should be able to get back to this week

**Comment by chinmaygarde** [comment]
> which internal bug are you referring to?

In the original issue description, you mentioned b/182520487 and b/182336500.

**Comment by chunhtai** [comment]
There is still one medium-priority internal bug, change to p3

**Comment by chunhtai** [comment]
proposed API will look like this: 
```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    const title = 'code sample';
    return MaterialApp(
      title: title,
      home: Scaffold(
        body: Home(),
      ),
    );
  }
}

class Home extends StatelessWidget {
  const Home({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          Semantics(
            attributedLabel: AttributedString('IwantThisSpellOut', attributes:[
              StringAttribute(range: TextRange(start: 0, end: 10), type:StringAttributeType.spellOut),
            ]),
            child: Text('IwantThisSpellOut'),
          ),
          Text('IwantThisSpellOut'),
          Text('IwantThisSpellOut'),
        ],
      ),
    );
  }
}
```

**Comment by chunhtai** [comment]
3 more tasks
1. Adding the API to TextSpan
2. Adding the API to TextField
3. Adding web support
